kinode/packages/kns_indexer/kns_indexer/src/lib.rs

```diff
@@ -41,15 +43,202 @@ const DELAY_MS: u64 = 1_000; // 1s
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
 struct State {
-    chain_id: u64,
-    // what contract this state pertains to
-    contract_address: eth::Address,
-    // namehash to human readable name
-    names: HashMap<String, String>,
-    // human readable name to most recent on-chain routing information as json
-    nodes: HashMap<String, net::KnsUpdate>,
+    // version of the state in kv
+    version: u32,
     // last block we have an update from
     last_block: u64,
+    // kv handle
+    // includes keys and values for:
+    // "meta:chain_id", "meta:version", "meta:last_block", "meta:contract_address",
+    // "names:{namehash}" -> "{name}", "nodes:{name}" -> "{node_info}"
+    kv: Kv<String, Vec<u8>>, // todo: maybe serialize directly into known enum of possible types?
```

Comment:
can you resolve this TODO before merge? what is the type going in here? it looks like just nodes, and then a handful of static types

Reply:
yep, the KV struct currently takes in a type for the value, I'll see if I can relax that requirement in the process_lib with a specific function passing in <K,V> rather than at instantiation. 

non-static-types right now: Strings for names and namehashes, net::KnsUpdates for node_infos

Reply:
using set_as::<u64> and get_as::<u64> reduced code amount here some, but left the default as Vec<u8> instead of having some bigger enum to match and serialize.

https://github.com/kinode-dao/process_lib/pull/123